qiling/os/memory.py

```diff
@@ -63,6 +63,29 @@ def __read_string(self, addr: int) -> str:
     def __write_string(self, addr: int, s: str, encoding: str):
         self.write(addr, bytes(s, encoding) + b'\x00')
 
+    def __getitem__(self, key: Union[slice, int]) -> bytearray:
+        if isinstance(key, slice):
+            start = key.start
+            stop = key.stop
+            step = key.step
+
+            if step is not None and step != 1:
+                # step != 1 means we have to do copy, don't allow it.
+                raise IndexError("Only support slicing continous memory")
+
+            return self.ql.mem.read(start, stop - start)
+        elif isinstance(key, int):
+            return self.ql.mem.read(key, 1)
```

Comment:
Since `mem.read` returns a `bytearray` it would be more intuitive to follow `bytearray` semantics here as well.
Perhaps return an integer value for a single index, like bytearrays do?

Reply:
done